### PR TITLE
Prevent file upload when folder creation failed

### PR DESCRIPTION
**Pull request by JammingBen**
## Description
We've fixed a bug where files would try to be uploaded if the creation of their respective folder failed beforehands.

Also improved the tests a bit:

* Converted the tests for `CreateAndUpload` to TypeScript
* Removed the upload helper tests from the component and moved them into a separate test suite

## Related Issue

- Fixes https://github.com/owncloud/web/issues/7957

## Screenshots

Upload failed

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingPublic2</strong> https://drone.owncloud.com/owncloud/web/29898/37/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUISharingPublicExpire-shareByPublicLinkExpiringLinks_feature-L62.png
  


</p></details>
:boom: The oC10SharingPublic2 tests pipeline failed. The build has been cancelled.

**Review by kulmann — APPROVED** [review]
Only a typo in the changelog item, I can change that in another PR when I prepare the next rc tag.

File: packages/web-runtime/tests/unit/composables/upload/useUpload.spec.ts [hunk 1]
```diff
@@ -10,6 +10,15 @@ import { SpaceResource } from 'web-client/src/helpers'
 import { mock } from 'jest-mock-extended'
 
 describe('useUpload', () => {
+  beforeEach(() => {
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    jest.spyOn(console, 'error').mockImplementation(() => {})
```
Comment (dschmidt) [inline packages/web-runtime/tests/unit/composables/upload/useUpload.spec.ts:15]:
```suggestion
    jest.spyOn(console, 'error').mockImplementation(() => undefined)
```

Then you don't need the eslint ignore

**Review by dschmidt — APPROVED** [review]

**Comment by ScharfViktor** [comment]
re-tested in rc2:

Screenshot 2022-11-28 at 18 03 04

**Comment by kulmann** [comment]
> re-tested in rc2:
> 
> Screenshot 2022-11-28 at 18 03 04

It's expected that this shows `Unknown error`. We can't bring new translation strings to the `stable-6.0` branch for technical reasons.

**Pull request merged**